src/Filter/StringFilter.php

```diff
@@ -30,29 +35,42 @@ class StringFilter extends Filter
         StringOperatorType::TYPE_EQUAL => '=',
     ];
 
+    /**
+     * Filtering types do not make sense for searching by empty value.
+     */
+    private const MEANINGLESS_TYPES = [
+        StringOperatorType::TYPE_CONTAINS,
+        StringOperatorType::TYPE_STARTS_WITH,
+        StringOperatorType::TYPE_ENDS_WITH,
+    ];
+
     public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $value)
     {
-        if (!$value || !\is_array($value) || !\array_key_exists('value', $value) || null === $value['value']) {
+        if (!$value || !\is_array($value) || !\array_key_exists('value', $value)) {
             return;
         }
 
-        $value['value'] = trim($value['value']);
+        $value['value'] = $this->trim($value['value'] ?? '');
+        $type = $value['type'] ?? StringOperatorType::TYPE_CONTAINS;
 
-        if (0 === \strlen($value['value'])) {
+        // ignore empty value if it doesn't make sense
+        if ('' === $value['value'] &&
+            (!$this->getOption('allow_empty') || \in_array($type, self::MEANINGLESS_TYPES, true))
+        ) {
             return;
         }
 
-        $type = $value['type'] ?? StringOperatorType::TYPE_CONTAINS;
         $operator = $this->getOperator((int) $type);
 
         // c.name > '1' => c.name OPERATOR :FIELDNAME
         $parameterName = $this->getNewParameterName($queryBuilder);
 
         $or = $queryBuilder->expr()->orX();
 
-        if ($this->getOption('case_sensitive')) {
+        if ($this->getOption('case_sensitive') || '' === $value['value']) {
```

Comment:
I understand why you add the check `|| '' === $value['value']`, but it could be the same for `' '`, `'     '`, `'0'`, '`42'`, `'$^&@'`, and lot of values so I find weird to have a special treatment here.